src/library_signals.js

```diff
@@ -138,20 +138,16 @@ var funs = {
     return -1;
   },
 #if SUPPORT_LONGJMP
-#if ASSERTIONS
-  siglongjmp__deps: ['longjmp'],
   siglongjmp: function(env, value) {
+#if ASSERTIONS
     // We cannot wrap the sigsetjmp, but I hope that
     // in most cases siglongjmp will be called later.
 
     // siglongjmp can be called very many times, so don't flood the stderr.
     warnOnce("Calling longjmp() instead of siglongjmp()");
-    _longjmp(env, value);
-  },
-#else
-  siglongjmp__sig: 'vii',
-  siglongjmp: 'longjmp',
```

Comment:
I first tried to change this to
```js
siglongjmp: '_emscripten_longjmp'
```
But then `wasm0.test_longjmp` succeeds but `wasm1.test_longjmp` fails saying "no alias for siglongjmp` [here](https://github.com/emscripten-core/emscripten/blob/ad438dfba750622d124f2de5b1154a374aeeaec4/src/modules.js#L270). Not really sure what's going on and why `wasm0.test_longjmp` works while `wasm1` doesn't, but all tests work if I move `if ASSERTIONS` into the function .

Reply:
You should probable move this alias into native code alongsize longjmp.  

If its really just an alias you can use `weak_alias` in the C source code, but it looks like the signature is different so you  might need to write a one-line wrapper here.

Reply:
That sounds better. I'll split this PR into two as you suggested and do it there.